Source/WebCore/page/ChromeClient.h

```diff
@@ -781,6 +781,13 @@ class ChromeClient {
     virtual void showCaptionDisplaySettings(CompletionHandler<void(bool)>&& callback) { callback(false); }
 #endif
 
+    using HardwareKeyboardAttachmentObserver = Function<void(bool)>;
+    virtual void addHardwareKeyboardAttachmentObserver(HardwareKeyboardAttachmentObserver&&) { }
+
+    virtual bool isHardwareKeyboardAttached() const { return true; }
```

Comment:
I think it's better without the `is` prefix
```suggestion
    virtual bool hardwareKeyboardAttached() const { return true; }
```